core/app/models/spree/zone.rb

```diff
@@ -15,17 +15,19 @@ class Zone < Spree::Base
     after_save :remove_defunct_members
     after_save :remove_previous_default
 
-    scope :with_member_ids,
-          ->(state_ids, country_ids) do
-            joins(:zone_members).where(
-              "(spree_zone_members.zoneable_type = 'Spree::State' AND
-                 spree_zone_members.zoneable_id IN (?))
-               OR (spree_zone_members.zoneable_type = 'Spree::Country' AND
-                 spree_zone_members.zoneable_id IN (?))",
-              state_ids,
-              country_ids
-            ).uniq
-          end
+    scope :with_member_ids, ->(state_ids, country_ids) do
+      return none unless state_ids || country_ids
```

Comment:
maybe this should check for `presence?` if `state_ids == []` and `country_ids == []` we know there will be no results